### PR TITLE
#249 statefulset custom labels

### PR DESCRIPTION
**Pull request by vktr-brlv**
Adding custom labels to managed cluster resources

### CODE REVIEW

**Comment by vktr-brlv** [comment]
Hey, @chrislovecnm !
I've added custom labels creation to statefulset. Do we need to pass them to pod resource?
Also I couldn't find annotations creation code, do we have one? Could you please point if we do?
Thanks!

**Comment by chrislovecnm** [comment]
@vktr-brlv we do not have code that does annotations.  Do we need it?

**Comment by vktr-brlv** [comment]
Not sure for now. I will remove annotations field.

What about passing all labels to pod not only NameKey, InstanceKey, ComponentKey?

**Comment by chrislovecnm** [comment]
We should pass all the labels to everything. Sts, pod, services, pdb, etc

File: pkg/labels/label.go [hunk 1]
```diff
@@ -43,6 +43,9 @@ var managed = []string{NameKey, InstanceKey, VersionKey, ComponentKey, PartOfKey
 
 func Common(cluster *api.CrdbCluster) Labels {
 	ll := Labels{}
+
+	Update(ll, cluster.Spec.Labels)
```
Comment (chrislovecnm) [inline pkg/labels/label.go:47]: We need a unit test for this and probably add a check on one of the e2e tests.

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Please add test coverage

File: api/v1alpha1/cluster_types.go [hunk 2]
```diff
@@ -86,6 +86,14 @@ type CrdbClusterSpec struct {
...
 	// +required
 	DataStore Volume `json:"dataStore,omitempty"`
+	// (Optional) Additional custom resource labels
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Map of additional custom labels"
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+	// (Optional) Additional custom resource annotations
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Map of additional custom annotations"
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
```
Comment (chrislovecnm) [inline api/v1alpha1/cluster_types.go:96]: Remove annotations for now please

**Comment by chrislovecnm** [comment]
@chrisseto PTAL

File: api/v1alpha1/cluster_types.go [hunk 3]
```diff
@@ -86,6 +86,14 @@ type CrdbClusterSpec struct {
 	// (Required) Database disk storage configuration
 	// +required
 	DataStore Volume `json:"dataStore,omitempty"`
+	// (Optional) Additional custom resource labels
```
Comment (chrisseto) [inline api/v1alpha1/cluster_types.go:89]: Could you note where these labels will appear? Are they on the statefulset and pods?

File: api/v1alpha1/cluster_types.go (hunk 2 above)
Comment (chrisseto) [reply api/v1alpha1/cluster_types.go:96]: Bump

**Review by chrisseto — CHANGES_REQUESTED** [review]

**Comment by chrislovecnm** [comment]
@vktr-brlv can you take a look at this, please?

**Comment by vktr-brlv** [comment]
Hi Chris, I'm working on it now.
Thanks!

File: api/v1alpha1/cluster_types.go (hunk 3 above)
Comment (vktr-brlv) [reply api/v1alpha1/cluster_types.go:91]: Hi @chrisseto ,
It should appear statefulset, pods, etc.
But currently in applies only to ss. I've updated both common and selector functions in labels.go with merge labels passed from crd. Could you please take a look and point what am I missing here? TIA

**Comment by keith-mcclellan** [comment]
@chrislovecnm is this something we can update and merge?

**Comment by chrislovecnm** [comment]
Close as I implement this in another PR

**Pull request closed without merging**